collagraph/component.py

```diff
@@ -0,0 +1,79 @@
+from abc import ABCMeta, abstractmethod
+
+from observ import reactive, readonly
+
+
+class Component(metaclass=ABCMeta):
+    """Abstract base class for components"""
+
+    def __init__(self, props=None):
+        self._props = readonly(props)
+        self._state = reactive({})
+        self._mounted = False
```

Comment:
I don't really follow why this `self._props` is private and readonly, and why `self._state` is private and why `self.state` is readonly?